README.md

```diff
@@ -53,7 +53,47 @@ cd onyo
 pip install -e ".[tests, docs]"
 ```
 
-### Tests
+### Design
+
+Onyo follows a set of development and coding conventions throughout its code
+base. For linting and type checking, the Onyo project uses flake8 and Pyre. The
+following sections describe additional conventions followed in this project,
+extending the existing [PEP 8 Style Guide](https://peps.python.org/pep-0008/).
+
+#### The Complete Code Base:
+- All classes, functions and properties have a doc-string describing their
+  functionality.
+- All functions have type hinting (including explicit return type for `None`).
+- The code conforms to [PEP 8](https://peps.python.org/pep-0008/).
+
+#### lib Tests (`tests/lib/`):
+- follow Onyo's code conventions, see [The Complete Code Base](#the-complete-code-base).
```

Comment:
This feels redundant, as the above is already "The Complete Code Base". Repeating in each section that "all" means "all" distracts.

If you're concerned that people won't consider tests to be code... then those people are just beyond our help. ;-)

Reply:
Well, that was at least my concern... :P